test/pipelinerun_test.go

```diff
@@ -35,175 +36,288 @@ import (
 			}
-		}
-	}
 
-	logger.Infof("Making sure events were created from taskrun and pipelinerun")
-	expectedNumberOfEvents := 3
-	// 1 from PipelineRun and 2 from Tasks defined in pipelinerun
+			if _, err := c.KubeClient.Kube.CoreV1().ServiceAccounts(namespace).Create(getPipelineRunServiceAccount(index, namespace)); err != nil {
+				t.Fatalf("Failed to create SA `%s`: %s", getName(hwSA, index), err)
+			}
 
-	matchKinds := map[string]string{"PipelineRun": "", "TaskRun": ""}
-	events := collectMatchingEvents(t, c.KubeClient, namespace, matchKinds, "Succeeded")
-	if len(events) != expectedNumberOfEvents {
-		t.Errorf("Expected %d number of successful events from pipelinerun and taskrun but got %d", expectedNumberOfEvents, len(events))
-	}
-}
+			pp := getHelloWorldPipelineParams(index, namespace)
+			// Set SA pipeline params spec
+			pp.Spec.ServiceAccount = getName(hwSA, index)
+			if _, err := c.PipelineParamsClient.Create(pp); err != nil {
+				t.Fatalf("Failed to create PipelineParams `%s`: %s", pp.Name, err)
+			}
 
-// Test pipeline trun which refers to service account.
-func TestPipelineRun_WithServiceAccount(t *testing.T) {
-	logger := logging.GetContextLogger(t.Name())
-	c, namespace := setup(t, logger)
+			if _, err := c.TaskClient.Create(&v1alpha1.Task{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: namespace,
+					Name:      getName(hwTaskName, index),
+				},
+				Spec: v1alpha1.TaskSpec{
+					// Reference build: https://github.com/knative/build/tree/master/test/docker-basic
+					BuildSpec: &buildv1alpha1.BuildSpec{
+						Steps: []corev1.Container{
+							corev1.Container{
+								Name:  "config-docker",
+								Image: "gcr.io/cloud-builders/docker",
+								// Private docker image for Build CRD testing
+								Command: []string{"docker"},
+								Args:    []string{"pull", "gcr.io/build-crd-testing/secret-sauce"},
+								VolumeMounts: []corev1.VolumeMount{{
+									Name:      "docker-socket",
+									MountPath: "/var/run/docker.sock",
+								}},
+							},
+						},
+						Volumes: []corev1.Volume{{
+							Name: "docker-socket",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/var/run/docker.sock",
+									Type: newHostPathType(string(corev1.HostPathSocket)),
+								},
+							},
+						}},
+					},
+				},
+			}); err != nil {
+				t.Fatalf("Failed to create Task `%s`: %s", getName(hwTaskName, index), err)
+			}
 
-	knativetest.CleanupOnInterrupt(func() { tearDown(t, logger, c, namespace) }, logger)
-	defer tearDown(t, logger, c, namespace)
+			if _, err := c.PipelineClient.Create(getHelloWorldPipelineWithSingularTask(index, namespace)); err != nil {
+				t.Fatalf("Failed to create Pipeline `%s`: %s", getName(hwPipelineName, index), err)
+			}
+		},
+		expectedTaskRuns:       []string{hwPipelineTaskName1},
+		expectedNumberOfEvents: 2,
+		// 1 from PipelineRun and 1 from Tasks defined in pipelinerun
```

Comment:
this is SO minor, but I'm used to seeing comments above the line they are commenting on?

(haha this PR is so great, I think this is about all I can come up with feedback wise XD)

Reply:
I have updated